wikimedia/wikipedia-ios
# Loosen successful http status code logic for article

**Phabricator:** 
https://phabricator.wikimedia.org/T299808
https://phabricator.wikimedia.org/T295194

### Notes
I believe this fixes the iOS15 article view audio bug. It turns out the `.ogg` request is returning an HTTP status code of 206 in iOS15, but in iOS14 it showed 200. We were strictly checking for a 200 status code, otherwise attempting to fall back to persisted files.

Note that this still does not play when in offline mode. I tracked this down to the `media-list` endpoint - although we do get an object representing this audio file, we do not get the full URL for download like we do with image types:

```
{
      "title": "File:Star-spangled_banner.ogg",
      "leadImage": false,
      "section_id": 19,
      "type": "audio",
      "audio_type": "generic",
      "showInGallery": false
}
```

So I believe this will require more followup work in the `media-list` endpoint, and this online fix isn't worth holding back on that.

I went back & forth a bit on how to structure this fix. The whole app could use an audit on hardcoded status codes, but I was aiming to keep this small, so I went with a small helper HTTPStatusCode struct. I'm open to an alternative form (or renaming) of this fix if there are suggestions.

### Test Steps
1. Go to Star Spangled Banner article on EN Wikipedia.
2. Scroll down to "Media" section and tap play on one of the audio bars.
3. Confirm audio plays while device is online on our supported iOS versions (iOS 13, 14, and 15). This bug was reproducible in simulator so a simulator check should be fine for PR review.
